packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/side-menu-panel.component.tsx

```diff
@@ -1,20 +1,36 @@
 import React, { useEffect } from 'react';
-import { LeftNavMenu, useOnClickOutside } from '@openmrs/esm-framework';
+import { isDesktop, LeftNavMenu, useLayoutType, useLeftNavStore, useOnClickOutside } from '@openmrs/esm-framework';
+import { createPortal } from 'react-dom';
 
 interface SideMenuPanelProps {
   expanded: boolean;
   hidePanel: Parameters<typeof useOnClickOutside>[0];
 }
 
+/**
+ * This is the menu that pops up when clicking on the hamburger button
+ * on the top nav. It's also responsible for rendering the left nav
+ * in desktop mode (via a react portal).
+ */
 const SideMenuPanel: React.FC<SideMenuPanelProps> = ({ expanded, hidePanel }) => {
   const menuRef = useOnClickOutside(hidePanel, expanded);
 
   useEffect(() => {
     window.addEventListener('popstate', hidePanel);
     return () => window.removeEventListener('popstate', hidePanel);
   }, [hidePanel]);
+  const layout = useLayoutType();
+  const { mode } = useLeftNavStore();
 
-  return expanded && <LeftNavMenu ref={menuRef} isChildOfHeader />;
+  const leftNavContainer = window.document.getElementById('omrs-left-nav-container-root');
+  return (
+    <>
+      {(!isDesktop(layout) || mode == 'collapsed') && expanded && <LeftNavMenu ref={menuRef} isChildOfHeader />}
+      {isDesktop(layout) &&
+        mode != 'collapsed' &&
+        createPortal(<LeftNavMenu ref={menuRef} isChildOfHeader />, leftNavContainer)}
```

Comment:
```suggestion
				leftNavContainer &&
        createPortal(<LeftNavMenu ref={menuRef} isChildOfHeader />, leftNavContainer)}
```

nit: Would it be useful to assert the existence of the `omrs-left-nav-container-root` DOM element here, or is that guaranteed?

Reply:
The `omrs-left-nav-container-root` div is hard-coded into `index.ejs`, so it should be guaranteed.